libs/server/ACL/AccessControlList.cs

```diff
@@ -98,7 +139,15 @@ public bool DeleteUser(string username)
             {
                 throw new ACLException("The special 'default' user cannot be removed from the system");
             }
-            return _users.TryRemove(username, out _);
+
+            bool userDeleted = _users.TryRemove(username, out _);
```

Comment:
There's a race here.

If Session A deletes a user and Session B adds the user it is possible for this to happen:

|Sessions A|Session B|
|--|--|
| `_users.TryRemove("foo", ...)` => true| |
| Thread Suspends | AddUser called with "foo", populates `_users`, does not modify `_userLockObjects` |
| Thread Resumes | |
| Because `userDeleted == true`, ` _userLockObjects.TryRemove("foo", out _)` called | |

At the end `_users` is populated, but `_userLockObject` is not.